temporal_capi/src/instant.rs

```diff
@@ -0,0 +1,117 @@
+#[diplomat::bridge]
+#[diplomat::abi_rename = "temporal_rs_{0}"]
+#[diplomat::attr(auto, namespace = "temporal_rs")]
+pub mod ffi {
+    use crate::duration::ffi::{Duration, TimeDuration};
+    use crate::error::ffi::TemporalError;
+    use crate::options::ffi::{DifferenceSettings, RoundingOptions};
+
+    #[diplomat::opaque]
+    pub struct Instant(pub temporal_rs::Instant);
+
+    /// For portability, we use two i64s instead of an i128.
+    /// The sign is extracted first before
+    /// appending the high/low segments to each other.
+    ///
+    /// This could potentially instead be a bit-by-bit split, or something else
+    pub struct I128Nanoseconds {
```

Comment:
question: would it be worthwhile defining a custom `i128` value for portability in the `primitive` module similar to `FiniteF64`?

Some context, I was already a bit concerned with using a `i128` on various platforms but had no hard evidence that this would be a blocker yet. Seeing this already brings up some general concerns I had, so I figured I'd see what your thoughts are.

Reply:
It's a problem over FFI more than a platform thing. C++ doesn't have uniform support, I think Rust is able to compile down to fallback variants if needed.

But potentially worth doing anyway.

Reply:
I think for now it might be fine in Rust, LLVM is smart about i128 even if it's a C++ extension.

We can change it if people ask us to.